src/data/nat/basic.lean

```diff
@@ -745,6 +745,12 @@ lemma lt_pow_self {p : ℕ} (h : 1 < p) : ∀ n : ℕ, n < p ^ n
   n + 1 < p^n + 1 : nat.add_lt_add_right (lt_pow_self _) _
     ... ≤ p ^ (n+1) : pow_lt_pow_succ h _
 
+lemma pow_monotonic { x m n : ℕ } ( ha : 2 ≤ x ) ( k : x^m ≤ x^n ) : m ≤ n :=
```

Comment:
Formatting nitpick: The convention is to not have spaces inside brackets: so `{x : X}` and not `{ x : X }`, also for `()` and `[]`, etc...

Also, this name is confusing. `pow_mono` suggest to me `m ≤ n` implies `x^m ≤ x^n`. This is going the other direction. I would call this `le_of_pow_le`.

There is a notion of `strict_mono` in the library. Once you prove that a function is strict mono, you can deduce `x ≤ y \iff f x ≤ f y` and similar things. In other words, it would apply `le_of_pow_le`.

Reply:
- Okay, thanks! I didn't see this in the style guide, maybe it should be added?
- Yes, I agree it's a poor name, thanks for the improvement.
- Perhaps I'm misunderstanding you, but this lemma will only apply in the case `2 ≤ x` so `^` isn't strict mono on the right in general

Reply:
> There is a notion of `strict_mono` in the library. Once you prove that a function is strict mono, you can deduce `x ≤ y \iff f x ≤ f y` and similar things. In other words, it would apply `le_of_pow_le`.

On further investigation, I'm inclined to agree. I'll try a bit more and see what the result looks like.